File: project_euler/Problem022/Problem022.java
package project_euler;

import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Problem022 {
    public void solution() throws IOException {
        String fileName = "names.txt";
        List<String> names = new ArrayList<>();

        try (BufferedReader handle = new BufferedReader(new InputStreamReader(new FileInputStream(fileName)))) {
            String line;
            while (null != (line = handle.readLine())) {
                names.add(line);
            }
        }

        Collections.sort(names);

        long result = 0L;
        for (int i = names.size() - 1; i >= 0; --i) {
            int score = 0;
            for (char letter : names.get(i).toCharArray()) {
                score += letter - 'A' + 1;
            }
            result += (long) score * (i + 1);
        }
        System.out.println(result);
    }
}

File: project_euler/Problem022/Problem022.js
function Problem022() {
    let fileName = 'names.txt'
    let names = []

    const file = require('n-readlines')
    let handle = new file(fileName)
    let line
    while (line = handle.next()) {
        names.push(line.toString())
    }

    names.sort(function(i, j) {return i.localeCompare(j)})

    let result = 0
    for (let [i, name] of names.entries()) {
        let score = 0
        for (let letter of name) {
            score += letter.charCodeAt(0) - 'A'.charCodeAt(0) + 1
        }
        result += score * (i + 1)
    }
    console.log(result)
}

module.exports = {
    Problem022 : Problem022
};
